storage/executable-store/src/lib.rs

```diff
@@ -0,0 +1,588 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_infallible::Mutex;
+use aptos_logger::{error, info};
+use aptos_types::executable::Executable;
+use dashmap::DashMap;
+use std::{
+    fmt::Debug,
+    hash::Hash,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+/// Represents the state of the ExecutableStore. Introduced to ensure the proper usage
+/// of the ExecutableStore throughout the system, as due to the current codebase structure
+/// constraints, different important steps are performed in different crates. Such
+/// defensive programming here ensures correctness (clearing the cache on any unexpected
+/// state updates) and observability to quickly identify any incorrect usage.
+///
+/// Initial state is Empty, and it is also possible to get to the Empty state by clearing
+/// the ExecutableStore at any time.
+///
+/// Execution in the system happens piece-wiese, where each piece contains a continuous
+/// sub-sequence of ledger of the ordered transactions. For example, in our codebase,
+/// pieces are called 'blocks' during the original execution / 'chunks' during state-sync.
+/// The only exception to this occurs while the cache in the Before(id) state is updated,
```

Comment:
only exception to what property?

Reply:
gosh, terrible comment. I missed a sentence there, most methods are to be called in quiescence, except ... Updated the comment.